src/components/Vitals.jsx

```diff
@@ -1,227 +1,29 @@
-
-                        const wcsUrl = `https://ows.rasdaman.org/rasdaman/ows?&SERVICE=WCS&VERSION=${version}&REQUEST=${request}&COVERAGEID=${coverage}&${timeSubset}&${xySubset}&FORMAT=${format}`;
-
-                        // submit WCS Request
-                        console.log("Fetching WCS Coverage:", wcsUrl);
-                        var xhr = new XMLHttpRequest();
-                        xhr.open('GET', wcsUrl, true);
-                        xhr.responseType = 'json';
-
-                        xhr.onload = function (event) {
-                            var wcsJSON = this.response;
-                            // handle WCS response
-                            if (wcsJSON) {
-                                const pixelValues = wcsJSON.flat(2);
-                                console.log(pixelValues);
-                                const maxValue = Math.max(...pixelValues);
-                                const minValue = Math.min(...pixelValues);
-                                const sum = pixelValues.reduce((acc, val) => acc + val, 0);
-                                const meanValue = sum / pixelValues.length;
-
-                                const prediction = (meanValue + (maxValue - meanValue) * 0.5);
-
-                                const newVitals = {
-                                    globalMin: minValue.toFixed(2),
-                                    globalMax: maxValue.toFixed(2),
-                                    globalAverage: meanValue.toFixed(2),
-                                    globalTrend: prediction.toFixed(2)
-                                }
-                                setVitals(newVitals);
-                            }
-                        };
-                        xhr.send();
-
-                    }
-                }, 10); // 10ms debounce, increase if needed
-            }
-
-        });
-    };
-
-    const removeSketchLayer = (globalVitals) => {
-        if (!mapView) {
-            return;
-        }
-
-        setVitals(globalVitals)
-
-        mapView.map.layers.items.forEach(layer => {
-            if (layer instanceof GraphicsLayer) {
-                mapView.map.remove(layer);
-            }
-        });
-
-        mapView.ui.components.forEach(sketch => {
-            if (sketch instanceof Sketch) {
-                sketch.destroy();
-            }
-        });
-
-        sketchRef.current.innerHTML = '';
-    };
-
-    useEffect(() => {
-        if (vitalSelection) {
-            setVitals(currentJSON.vitals);
-            setLayerName({ layerName: currentJSON.name, layerUnit: currentJSON.unit });
-            addSketchLayer();
-        } else {
-            removeSketchLayer(currentJSON.vitals);
-        }
-    }, [mapView, vitalSelection, currentJSON]);
+    const { vitalsData } = useContext(VitalsDataContext);
+    const globalAverage = vitalsData?.globalAverage ?? 'N/A';
+    const globalMax = vitalsData?.globalMax ?? 'N/A';
```

Comment:
We're setting the default values for these in 3 different places. Is that necessary?